### DISCUSSION TITLE
Is there a convinient way to get cookies from ReqHeader?

### DISCUSSION THREAD

**Discussion by ModProg** [actix/actix-web#2504] [Q&A] [answered]
I know that I can parse them manually, but my question is, is there a convenience method like `HttpRequest::cookie(&str)`, to get them from `RequestHead` in a guard?
And if not, would that be something to consider adding, maybe in a trait both implement?

**Comment by ModProg** [comment]
Well NVM I cannot use it the way I wanted anyway, because I cannot query the database in a guard.

**Comment by robjtede** [comment, marked as answer]
It's not convenient to get cookies in a guard with current methods. Tracking feature addition here: https://github.com/actix/actix-web/issues/2575
